cmd/infra/aws/iam.go

```diff
@@ -686,6 +655,51 @@ func (o *CreateIAMOptions) CreateWorkerInstanceProfile(client iamiface.IAMAPI, p
 	return nil
 }
 
+type CreateIAMRoleOptions struct {
+	RoleName          string
+	TrustPolicy       string
+	PermissionsPolicy string
+
+	additionalIAMTags []*iam.Tag
```

Comment:
why is this not exported?

Reply:
same as https://github.com/openshift/hypershift/pull/4027#discussion_r1599624238